reactor-pool/src/main/java/reactor/pool/SimpleDequePool.java

```diff
@@ -442,13 +446,15 @@ else if (sig.isOnError()) {
 							logger.debug("should warm up {} extra resources", toWarmup);
 
 							final long startWarmupIteration = clock.millis();
-							Flux<Void> warmupFlux = Flux.range(1, toWarmup)
-							    //individual warmup failures decrement the permit and are logged
-							    .flatMap(i -> warmupMono(i, toWarmup, startWarmupIteration, allocator));
-
-							primary.onErrorResume(e -> Mono.empty())
-							       .thenMany(warmupFlux)
-							       .subscribe(aVoid -> { }, alreadyPropagatedOrLogged -> drain(), this::drain);
+							Flux<Mono<POOLABLE>> monos = Flux.range(0, toWarmup + 1)
+									.map(n -> (n == 0) ? primary : warmupMono(n, toWarmup, startWarmupIteration, allocator));
+
+							// merge will eagerly subscribe to the allocator from the current thread, but the concurrency
+							// can be controlled from configuration
+							int mergeConcurrency = Math.min(poolConfig.allocationStrategy().warmupParallelism(), toWarmup + 1);
+							Flux.merge(monos, mergeConcurrency, Queues.XS_BUFFER_SIZE)
+									.onErrorResume(e -> Mono.empty())
+									.subscribe(poolable -> drain(), alreadyPropagatedOrLogged -> drain(), () -> drain());
```

Comment:
There is a misscalculation, it seems that extra resource is warmed up. Lets keep code as it was before and add primary to maralle merge in a different way

```suggestion
							// flatMap will eagerly subscribe to the allocator from the current thread, but the concurrency
							// can be controlled from configuration
							final int mergeConcurrency = Math.min(poolConfig.allocationStrategy().warmupParallelism(), toWarmup + 1);
							Flux.range(1, toWarmup)
								.map(i -> warmupMono(i, toWarmup, startWarmupIteration, allocator).doOnSuccess(__ -> drain()))
								.startWith(primary.doOnSuccess(__ -> drain()).then())
								.flatMap(Function.identity(), mergeConcurrency, 1) // since we dont store anything the inner buffer can be simplified
								.onErrorResume(e -> Mono.empty())
								.subscribe(aVoid -> { }, alreadyPropagatedOrLogged -> drain(), this::drain)
```

Reply:
Applied feedback (thanks !), see applied feedback, see https://github.com/reactor/reactor-pool/pull/171/commits/c6ddf8b593f8504711bccaa51484e8761c48c317